File: test/examples/java/Annotations.java
package examples.java;

import examples.java.support.FancyAnnotation;
import examples.java.support.MarkerAnnotation;

@Deprecated
@FancyAnnotation
@MarkerAnnotation(intProperty = 456, stringProperty = "class")
public class Annotations {

    @FancyAnnotation
    @MarkerAnnotation(intProperty = 789, stringProperty = "method")
    public void methodAndParameterAnnotationTest(int x, @MarkerAnnotation(intProperty = 123, stringProperty = "parameter") long y, String s) {
    }
}

File: test/examples/kotlin/Annotations.kt
package examples.kotlin

annotation class fancy
annotation class nancy (val why : String, val me : Int)

fancy nancy("class", 1) class Annotations {
    fancy nancy("method", 2) fun annotatedMethod(o : Long, fancy nancy("parameter", 3) annotatedParameter : Int, s : String): Int {
        return 4
    }
}